src/main/java/Controller/userController.java

```diff
@@ -1,4 +1,42 @@
 package Controller;
 
+import InputBoundary.loginInputBoundary;
+import InputBoundary.RegisterInputBoundary;
+import InputData.loginInputData;
+
+
+/**
+ * Controller for login and register use cases.
+ */
 public class userController {
+    private final loginInputBoundary loginInteractor;
+    private final RegisterInputBoundary registerInteractor;
+    private final loginInputData loginInputData;
+    private final loginInputData registerInputData;
+    /**
+     * Constructor.
+     * @param loginInteractor use case interactor for adding a song to the space
+     * @param registerInteractor use case interactor for playing the space
+     */
+    public userController(loginInputBoundary loginInteractor, RegisterInputBoundary registerInteractor,
+                          String username, String password, boolean isArtist) {
+        this.loginInteractor = loginInteractor;
```

Comment:
IIRC the controller should construct the loginInteractor -- your code implies that the ViewModel creates it. Technically, it's fine if the ViewModel does this (as they're in the same layer), but I'd prefer we stay consistent across use cases. 

Same applies for registerInteractor.

Thanks for adding Interactor as a suffix to the use case files, btw!

Reply:
Ic, I'll add that in rn. Thanks!